src/util/rate_limit.rs

```diff
@@ -0,0 +1,167 @@
+use conduit::Request;
+use std::collections::HashMap;
+
+use crate::db::{DieselPool};
+use crate::middleware::app::RequestApp;
+use crate::models::{User};
+use crate::util::errors::{CargoResult, TooManyRequests};
+use std::sync::{Arc, Mutex};
+
+use chrono::{DateTime, Duration, Utc};
+
+/// Settings for a rate-limited route.
+#[derive(Debug, Clone)]
+pub struct RateLimitSettings {
+    /// The code for this category. Can be stored in a database, etc.
+    pub key: String,
+    /// The maximum number of tokens that can be acquired.
+    pub max_amount: usize,
+    /// How often we refill
+    pub refill_time: Duration,
+    /// The number of tokens that are added during a refill.
+    pub refill_amount: usize,
```

Comment:
I don't think we need to make this configurable. It'll always be 1 for the foreseeable future.